orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt

```diff
@@ -56,4 +57,23 @@ interface AuthenticationAware {
       ExecutionContext.clear()
     }
   }
+  fun backtrackSkippedStages(stage: StageExecution): StageExecution {
+    if (stage.type == "manualJudgment" &&
```

Comment:
I have an idea how we could approach such checks in a slightly different way - not better, probably not worse, just different. 
There's the Demeter law that says that a module should not have the knowledge on the inner details of the objects it manipulates. In this case we need to get access to internal details (like type or status) and then use them in checks like stage.type == "X". We could replace it with methods like stage.isType("X") or stage.isStatus("S"). It's very convenient to use and it is future-proof. If one day stage.type implementation changes (now type is a String, but in the future it might be a designated new class like Type), checks like stage.type == "X" will need to be updated everywhere to match stage.type new class. If we create a convenient method like stage.isType("X"), we can make adjustments to new type class only inside that method. 

Just a food for thought :) I am fine with your changes as they are now!

Reply:
Nice though @chriskotula I applied your recommendation, It will let us keep a maintainable code for possible future changes. Thanks!